src/ontodia/viewUtils/toSvg.ts

```diff
@@ -278,38 +308,53 @@ export function toDataURL(options: ToSVGOptions & ToDataURLOptions): Promise<str
         const contentWidth = imageRect.width - 2 * padding;
         const contentHeight = imageRect.height - 2 * padding;
 
-        const img = new Image();
-        img.onload = function () {
-            let dataURL: string;
-            let context: CanvasRenderingContext2D;
-            let canvas: HTMLCanvasElement;
-
-            function createCanvas() {
-                canvas = document.createElement('canvas');
-                canvas.width = imageRect.width;
-                canvas.height = imageRect.height;
-                context = canvas.getContext('2d');
-                context.fillStyle = options.backgroundColor || 'white';
-                context.fillRect(0, 0, imageRect.width, imageRect.height);
-            }
+        const svgOptions = {...options, convertImagesToDataUris: true, mockImages: isIE11() };
+        svgOptions.convertImagesToDataUris = true;
 
-            createCanvas();
-            try {
-                context.drawImage(img, padding, padding, contentWidth, contentHeight);
-                dataURL = canvas.toDataURL(mimeType, options.quality);
-                resolve(dataURL);
-            } catch (e) {
-                reject(e);
-                return;
-            }
-        };
-        const svgOptions = {...options, convertImagesToDataUris: true};
-        toSVG(svgOptions).then(svgString => {
-            svgString = svgString
-                .replace('width="100%"', 'width="' + contentWidth + '"')
-                .replace('height="100%"', 'height="' + contentHeight + '"');
-            img.src = 'data:image/svg+xml,' + encodeURIComponent(svgString);
-        });
+        const { canvas, context } = createCanvas();
+
+        if (!isIE11()) {
+            const img = new Image();
+            img.onload = function () {
+                try {
+                    context.drawImage(img, padding, padding, contentWidth, contentHeight);
+                    resolve(canvas.toDataURL(mimeType, options.quality));
+                } catch (e) {
+                    reject(e);
+                    return;
+                }
+            };
+
+            toSVG(svgOptions).then(svgString => {
+                svgString = svgString
+                    .replace('width="100%"', 'width="' + contentWidth + '"')
```

Comment:
Use `preserveDimensions` option of `toSVG()` instead of string replacement hack.